handlers/user.go

```diff
@@ -1727,8 +1728,8 @@ func internUserNotificationsSubscribe(event, filter string, threshold float64, w
 
 	errFields["event_name"] = eventName
 
-	if !isValidSubscriptionFilter(eventName, filter) {
-		utils.LogError(nil, "error invalid filter: not pubkey or client for subscription", 0, errFields)
+	if valid, err := isValidSubscriptionFilter(user.UserID, eventName, filter); err != nil || !valid {
+		utils.LogError(err, "error invalid filter: not pubkey or client for subscription", 0, errFields)
 		ErrorOrJSONResponse(w, r, "Internal server error", http.StatusInternalServerError)
```

Comment:
**Issue**: An internal server error should only be returned if `err != nil` otherwise it should be `400 Bad Request` (or `404` if you implement my previous comment).
This also goes for other calling instances of `isValidSubscriptionFilter`

Reply:
Improved api error